### PR TITLE
DMU 8 Cards + 2 Fix

### PR DESCRIPTION
**Pull request by Simisays**
Wasn't able to fix Monstrous War-Leech feel free to comment if you have ideas. It's working correctly but doesn't display PT correctly outside of battlefield

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/braids_arisen_nightmare.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ ManaCost:1 B B
 Types:Legendary Creature Nightmare
 PT:3/3
 T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigSac | TriggerDescription$ At the beginning of your end step, you may sacrifice an artifact, creature, enchantment, land, or planeswalker. If you do, each opponent may sacrifice a permanent that shares a card type with it. For each opponent who doesn't, that player loses 2 life and you draw a card.
-SVar:TrigSac:DB$ Sacrifice | Optional$ True | SacValid$ Artifact,Creature,Enchantment,Land,Planeswalker | Amount$ 1 | RememberSacrificed$ True | SubAbility$ DBRepeatEach
+SVar:TrigSac:DB$ Sacrifice | OptionalDecider$ You | SacValid$ Artifact,Creature,Enchantment,Land,Planeswalker | Amount$ 1 | RememberSacrificed$ True | SubAbility$ DBRepeatEach
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/braids_arisen_nightmare.txt:6]: This change is wrong

File: forge-gui/res/cardsfolder/upcoming/scout_the_wilderness.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Name:Scout the Wilderness
+ManaCost:2 G
+Types:Sorcery
+K:Kicker:1 W
+A:SP$ ChangeZone | Origin$ Library | Destination$ Battlefield | ChangeType$ Land.Basic | ChangeNum$ 1 | SubAbility$ DBToken | SpellDescription$ Search your library for a basic land card, put it onto the battlefield tapped, then shuffle. If this spell was kicked, create two 1/1 white Soldier creature tokens.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/scout_the_wilderness.txt:5]: Missing Tapped

File: forge-gui/res/cardsfolder/upcoming/shadow_prophecy.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Name:Shadow Prophecy
+ManaCost:2 B
+Types:Instant
+A:SP$ Dig | DigNum$ X | ChangeNum$ 2 | SubAbility$ DBLoseLife | SpellDescription$ Domain — Look at the top X cards of your library, where X is the number of basic land types among lands you control. Put up to two of them into your hand and the rest into your graveyard. You lose 2 life.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/shadow_prophecy.txt:4]: `Optional$ True`

Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/shadow_prophecy.txt:4]: Missing Graveyard

File: forge-gui/res/cardsfolder/upcoming/shield_wall_sentinel.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Name:Shield-Wall Sentinel
+ManaCost:4
+Types:Artifact Creature Golem
+PT:1/3
+K:Defender
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ DBSearch | TriggerDescription$ When CARDNAME enters the battlefield, you may search your library for a creature card with defender, reveal it, put it into your hand, then shuffle.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/shield_wall_sentinel.txt:6]: `OptionalDecider$ You`

File: forge-gui/res/cardsfolder/upcoming/shield_wall_sentinel.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+SVar:DBSearch:DB$ ChangeZone | Origin$ Library | Destination$ Hand | ChangeType$ Creature.withDefender | ChangeNum$ 1
+DeckNeeds:Type|Defender
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/shield_wall_sentinel.txt:8]: Malformed entries

Also Defender is not a type but a KW

File: forge-gui/res/cardsfolder/upcoming/soul_of_windgrace.txt [hunk 6]
```diff
@@ -0,0 +1,14 @@
+Name:Soul of Windgrace
+ManaCost:1 B R G
+Types:Legendary Creature Cat Avatar
+PT:5/4
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigChangeZone | TriggerDescription$ Whenever CARDNAME enters the battlefield, you may put a land card from a graveyard onto the battlefield tapped under your control.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/soul_of_windgrace.txt:5]: those triggers are optional again

File: forge-gui/res/cardsfolder/upcoming/braids_frightful_return.txt [hunk 7]
```diff
@@ -2,10 +2,10 @@ Name:Braids's Frightful Return
 ManaCost:2 B
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/braids_frightful_return.txt:2]: I think the file name should be `braidss_frightful_return.txt` as ugly as that looks

File: forge-gui/res/cardsfolder/upcoming/scout_the_wilderness.txt [hunk 8]
```diff
@@ -0,0 +1,9 @@
+Name:Scout the Wilderness
+ManaCost:2 G
+Types:Sorcery
+K:Kicker:1 W
+A:SP$ ChangeZone | Origin$ Library | Destination$ Battlefield | Tapped$ True | ChangeType$ Land.Basic | ChangeNum$ 1 | SubAbility$ DBToken | SpellDescription$ Search your library for a basic land card, put it onto the battlefield tapped, then shuffle. If this spell was kicked, create two 1/1 white Soldier creature tokens.
+SVar:DBToken:DB$ Token | TokenAmount$ 2 | TokenScript$ w_1_1_soldier | Condition$ Kicked
+DeckHints:Color$White
+DeckHas:Ability$Token
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/scout_the_wilderness.txt:8]: ` & Type$Soldier`

File: forge-gui/res/cardsfolder/upcoming/sheoldreds_restoration.txt [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+ManaCost:3 B
+Types:Sorcery
+K:Kicker:2 W
+A:SP$ ChangeZone | Origin$ Graveyard | Destination$ Battlefield | ValidTgts$ Creature.YouOwn | SubAbility$ DBGainLife | TgtPrompt$ Select target creature to return from your graveyard |  TriggerDescription$ Return target creature card from your graveyard to the battlefield. If this spell was kicked, you gain life equal to that card's mana value. Otherwise, you lose that much life.
+SVar:DBGainLife:DB$ GainLife | LifeAmount$ X | SubAbility$ DBLoseLife | Condition$ Kicked
+SVar:DBLoseLife:DB$ LoseLife | LifeAmount$ X | ConditionCheckSVar$ Y | ConditionSVarCompare$ EQ0 | SubAbility$ DBChange
+SVar:DBChange:DB$ ChangeZone | Origin$ Stack | Destination$ Exile
+SVar:X:Targeted$CardManaCost
+SVar:Y:Count$TimesKicked
+DeckHas:Ability$Graveyard|Lifegain
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/sheoldreds_restoration.txt:11]: `Lifegain` > `LifeGain`

File: forge-gui/res/cardsfolder/upcoming/vodalian_mindsinger.txt [hunk 10]
```diff
@@ -2,7 +2,7 @@ Name:Vodalian Mindsinger
 ManaCost:1 U U
 Types:Creature Merfolk Wizard
 PT:2/2
-K:Kicker:1 B:1 G
+K:Kicker:1 R:1 G
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/vodalian_mindsinger.txt:5]: probably add `DeckHints:Color$Red|Green` too

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/vodalian_mindsinger.txt:5]: also fix `’` > `'` in Oracle
also add `.` to end of line 6

File: forge-gui/res/cardsfolder/upcoming/sheoldreds_restoration.txt [hunk 11]
```diff
@@ -0,0 +1,12 @@
+Name:Sheoldred's Restoration
+ManaCost:3 B
+Types:Sorcery
+K:Kicker:2 W
+A:SP$ ChangeZone | Origin$ Graveyard | Destination$ Battlefield | ValidTgts$ Creature.YouOwn | SubAbility$ DBGainLife | TgtPrompt$ Select target creature to return from your graveyard |  TriggerDescription$ Return target creature card from your graveyard to the battlefield. If this spell was kicked, you gain life equal to that card's mana value. Otherwise, you lose that much life.
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/sheoldreds_restoration.txt:5]: Needs at the end:
Exile CARDNAME.

File: forge-gui/res/cardsfolder/upcoming/soul_of_windgrace.txt [hunk 12]
```diff
@@ -0,0 +1,14 @@
+Name:Soul of Windgrace
+ManaCost:1 B R G
+Types:Legendary Creature Cat Avatar
+PT:5/4
+T:Mode$ ChangesZone | Origin$ Any | OptionalDecider$ You | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigChangeZone | TriggerDescription$ Whenever CARDNAME enters the battlefield, you may put a land card from a graveyard onto the battlefield tapped under your control.
+T:Mode$ Attacks | ValidCard$ Card.Self | OptionalDecider$ You | Execute$ TrigChangeZone | TriggerDescription$ Whenever CARDNAME attacks, you may put a land card from a graveyard onto the battlefield tapped under your control.
+SVar:TrigChangeZone:DB$ ChangeZone | Origin$ Graveyard | Destination$ Battlefield | GainControl$ True | Tapped$ True | TgtPrompt$ Select target land card in a graveyard | ValidTgts$ Land
+A:AB$ GainLife | Cost$ G Discard<1/Land> | LifeAmount$ 3 | SpellDescription$ Gain 3 Life
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/soul_of_windgrace.txt:8]: SpellDescription$ You gain 3 life.

**Review by paulsnoops — APPROVED** [review]

File: forge-gui/res/cardsfolder/upcoming/sheoldreds_restoration.txt [hunk 13]
```diff
@@ -0,0 +1,12 @@
+Name:Sheoldred's Restoration
+ManaCost:3 B
+Types:Sorcery
+K:Kicker:2 W
+A:SP$ ChangeZone | Origin$ Graveyard | Destination$ Battlefield | ValidTgts$ Creature.YouOwn | SubAbility$ DBGainLife | TgtPrompt$ Select target creature to return from your graveyard |  TriggerDescription$ Return target creature card from your graveyard to the battlefield. If this spell was kicked, you gain life equal to that card's mana value. Otherwise, you lose that much life. Exile CARDNAME.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/sheoldreds_restoration.txt:5]: ` |..TriggerDescription$` > ` |.TriggerDescription$`

File: forge-gui/res/cardsfolder/upcoming/soul_of_windgrace.txt [hunk 14]
```diff
@@ -0,0 +1,14 @@
+Name:Soul of Windgrace
+ManaCost:1 B R G
+Types:Legendary Creature Cat Avatar
+PT:5/4
+T:Mode$ ChangesZone | Origin$ Any | OptionalDecider$ You | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigChangeZone | TriggerDescription$ Whenever CARDNAME enters the battlefield, you may put a land card from a graveyard onto the battlefield tapped under your control.
+T:Mode$ Attacks | ValidCard$ Card.Self | OptionalDecider$ You | Execute$ TrigChangeZone | TriggerDescription$ Whenever CARDNAME attacks, you may put a land card from a graveyard onto the battlefield tapped under your control.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/soul_of_windgrace.txt:6]: add `Secondary$ True`, then make the TriggerDescription$ here and in line 5 identical to the Oracle

**Review by paulsnoops — APPROVED** [review]

**Pull request merged**